src/compiler/parse_table.h

```diff
@@ -97,6 +97,8 @@ class ParseTable {
 
   std::vector<ParseState> states;
   std::map<rules::Symbol, ParseTableSymbolMetadata> symbols;
+
+  std::set<rules::Symbol> mergeable_symbols;
```

Comment:
I think it does make sense to call them **mergeable**, because when considering which symbols to merge, if we encounter any symbol that is **not** in this set, we abort merging:

```cpp
if (mergeable_symbols.count(symbol) == 0)
  return false;
```

Thus, it functions as a whitelist of symbols that *can* be merged.